portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileShortcutServiceImpl.java

```diff
@@ -88,6 +88,27 @@ public void deleteFileShortcut(String externalReferenceCode, long groupId)
 		dlFileShortcutLocalService.deleteFileShortcut(fileShortcut);
 	}
 
+	@Override
+	public DLFileShortcut getDLFileShortcutByExternalReferenceCode(
+			String externalReferenceCode, long groupId)
+		throws PortalException {
+
+		DLFileShortcut fileShortcut =
+			dlFileShortcutLocalService.getDLFileShortcutByExternalReferenceCode(
+				externalReferenceCode, groupId);
+
+		ModelResourcePermission<FileShortcut>
+			fileShortcutModelResourcePermission =
+				ModelResourcePermissionRegistryUtil.getModelResourcePermission(
+					FileShortcut.class.getName());
+
+		fileShortcutModelResourcePermission.check(
+			getPermissionChecker(), fileShortcut.getFileShortcutId(),
+			ActionKeys.VIEW);
+
+		return fileShortcut;
+	}
+
```

Comment:
Why have `DLFileShortcut` in the name instead of simply `FileShortcut`? The rest of the methods in this class don't have the "DL" prefix.